discovery/sync_manager_test.go

```diff
@@ -226,14 +227,29 @@ func TestSyncManagerRotateActiveSyncerCandidate(t *testing.T) {
 	assertPassiveSyncerTransition(t, passiveSyncer, passiveSyncPeer)
 }
 
+// TestSyncManagerNoInitialHistoricalSync ensures no initial sync is attempted
+// when NumActiveSyncers is set to 0.
+func TestSyncManagerNoInitialHistoricalSync(t *testing.T) {
+	t.Parallel()
+
+	syncMgr := newTestSyncManager(0)
+	syncMgr.Start()
+	defer syncMgr.Stop()
+
+	// We should not expect any messages from the peer.
+	peer := randPeer(t, syncMgr.quit)
+	syncMgr.InitSyncState(peer)
+	assertNoMsgSent(t, peer)
```

Comment:
We should also test that the `HistoricalSyncTicker` doesn't tick and forces a historical sync after the expected time.

Reply:
I addressed this in a different way. I gated the `forceHistoricalSync` logic with a check on the number of desired active syncers. In the test we force a historical sync and then check that nothing happens. This way even if the timer is accidentally set, nothing should happen. lmk wyt